pkg/kubeapiserver/authorizer/config.go

```diff
@@ -172,9 +174,9 @@ func LoadAndValidateData(data []byte, requireNonWebhookTypes sets.Set[authzconfi
 	}
 
 	// validate the file and return any error
-	if errors := validation.ValidateAuthorizationConfiguration(nil, authorizationConfiguration,
-		sets.NewString(modes.AuthorizationModeChoices...),
-		sets.NewString(repeatableAuthorizerTypes...),
+	if errors := validation.ValidateAuthorizationConfiguration(authorizationcel.NewDefaultCompiler(), nil, authorizationConfiguration,
```

Comment:
same question about whether to plumb or construct new ... it seems confusing to use one compiler to validate the config and a different compiler (held by reloadableAuthorizerResolver) to construct the authorizer